pandas/tests/strings/test_find_replace.py

```diff
@@ -939,13 +940,22 @@ def test_find_bad_arg_raises(any_string_dtype):
         ser.str.rfind(0)
 
 
-def test_find_nan(any_string_dtype):
+def test_find_nan(any_string_dtype, request):
     ser = Series(
         ["ABCDEFG", np.nan, "DEFGHIJEF", np.nan, "XXXX"], dtype=any_string_dtype
     )
     expected_dtype = (
         np.float64 if is_object_or_nan_string_dtype(any_string_dtype) else "Int64"
     )
+    if (
+        pa_version_under13p0
+        and isinstance(ser.dtype, pd.StringDtype)
+        and ser.dtype.storage == "pyarrow"
+    ):
+        # https://github.com/apache/arrow/issues/36311
+        mark = pytest.mark.xfail(reason="https://github.com/apache/arrow/issues/36311")
+        # raises pa.lib.ArrowInvalid with Negative buffer resize
+        request.node.add_marker(mark)
```

Comment:
Alternative could be to check the pyarrow version in the implementation, and in that case for older pyarrow use the python implementation? 

(I don't know whether this happens in general or only for very specific cases. Based on looking at the test, this is buggy whenever there are missing values? In theory could also check for that, and only then fallback)

Reply:
sure, updated